src/pages/Polling.js

```diff
@@ -112,140 +109,416 @@ const DetailsCardText = styled.p`
+            )}
+            renderRowItem={item => <DropdownText>{item}</DropdownText>}
+            value={selectedOption || optionVotingFor || 'Please choose...'}
+            onSelect={this.onDropdownSelect}
+            emptyMsg="Not available"
+          />
+          <VoteButton
+            bgColor="green"
+            color="white"
+            hoverColor="white"
+            width="135px"
+            disabled={
+              !poll.active ||
+              !activeAccount ||
+              selectedOptionId === undefined ||
+              selectedOption === optionVotingFor
+            }
+            onClick={() =>
+              modalOpen(PollingVote, {
+                poll: {
+                  pollId,
+                  selectedOption,
+                  selectedOptionId,
+                  totalVotes
+                }
+              })
+            }
+          >
+            Vote Now
+          </VoteButton>
+        </VoteSelection>
+      </React.Fragment>
+    );
+  }
 }
 
-const reduxProps = ({ proposals, tally, accounts, metamask }, { match }) => {
+const timeLeft = (startDate, endDate) => {
+  const timeLeft = Math.floor(endDate / 1000) - Math.floor(startDate / 1000);
+  const days = Math.floor(timeLeft / (3600 * 24));
+  return days !== 1 ? `${days} days` : `${days} day`;
+};
+
+const downloadRawPollData = (multiHash, rawData) => {
+  const element = document.createElement('a');
+  const file = new Blob([rawData], { type: 'text/plain' });
+  element.href = URL.createObjectURL(file);
+  element.download = `${multiHash}.txt`;
+  document.body.appendChild(element);
+  element.click();
+};
+
+class Polling extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeAccount: this.props.activeAccount,
+      voteStateFetching: true
+    };
+  }
+
+  static getDerivedStateFromProps(newProps, state) {
+    /* Replaces componentWillReceiveProps
+    https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#fetching-external-data-when-props-change */
+
+    if (newProps.activeAccount !== state.activeAccount) {
+      return { activeAccount: newProps.activeAccount };
+    } else return null;
+  }
+
+  async componentDidUpdate(prevProps) {
+    if (
+      this.state.activeAccount !== prevProps.activeAccount ||
+      (this.props.poll && prevProps.poll === undefined)
+    ) {
+      await this.updateVotedPollOption();
+    }
+  }
+
+  updateVotedPollOption = async () => {
+    if (!this.props.poll || !this.state.activeAccount) return null;
+    await this.props.getOptionVotingFor(
+      this.state.activeAccount.address,
+      this.props.poll.pollId
+    );
+    this.setState({
+      voteStateFetching: false
+    });
+  };
+
+  render() {
+    const { activeAccount, voteStateFetching } = this.state;
+    const {
+      poll,
+      isValidRoute,
+      network,
+      accountDataFetching,
+      modalOpen
+    } = this.props;
+    if (!poll) return null;
```

Comment:
@b-pmcg would it make sense to return a `NotFound` page here?

Reply:
The intention was just to not finish `render()` if `poll` hasn't been populated yet, but we expect that it should be. Mostly because the component pretty much relies on `poll` for everything. This was throwing errors when landing directly on a poll's detail page before we had time to populate everything.

We return `NotFound` a few lines down if the poll doesn't exist and/or its an invalid route. I'm not sure if its the best way to protect against async data that hasn't arrived yet, let me know if there's a more clear way.

Reply:
gotcha, there are specialized components we could use, but i think what we have is ok. I only ask b/c i was seeing an empty page when hitting a poll that didn't exist rather than `not found`. might see if there's an easy tweak for that

Reply:
I see what you mean, I'll try to find a better way to distinguish the two states.